src/scalarstats.jl

```diff
@@ -37,6 +37,28 @@ function harmmean(a::RealArray)
     return n / s
 end
 
+# Generalized mean
+"""
+    genmean(a, p)
+
+Return the generalized/power mean with exponent `p` of a real-valued array,
+i.e. ``\left( \frac{1}{n} \sum_{i=1}^n a_i^p \right)^{\frac{1}{p}}``, where `n = length(a)`.
+It is taken to be the geometric mean when `p == 0`.
+"""
+function genmean(a::RealArray, p::Real)
+  if p == 0
+    return geomean(a)
+  end
+  s = 0.0
+  n = length(a)
+  for x in a
+    #= At least one of `x` or `p` must not be an int to avoid domain errors when `p` is a negative int.
+    We choose `x` in order to exploit exponentiation by squaring when `p` is an int. =#
+    @inbounds s += convert(Float64, x)^p
+  end
+  return (s/n)^(1/p)
```

Comment:
Please indent with 4 spaces rather than 2 for consistency with the rest of the repo